core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -879,8 +879,19 @@ private boolean checkConsistencyInternal(Inode inode, AlluxioURI path)
       return ufs.isDirectory(ufsPath);
     } else {
       InodeFile file = (InodeFile) inode;
-      return ufs.isFile(ufsPath)
-          && ufs.getFileStatus(ufsPath).getContentLength() == file.getLength();
+      if (ufs.isFile(ufsPath)) {
+        UfsFileStatus ufsFileStatus = ufs.getFileStatus(ufsPath);
+        if (ufs.getFileStatus(ufsPath).getContentLength() != file.getLength()) {
+          return false;
+        } else if (hasAnyBlockCached(mBlockMaster.getBlockInfoList(file.getBlockIds()))
+            && (ufsFileStatus.getLastModifiedTime() > file.getLastModificationTimeMs())) {
```

Comment:
The UFS clock timing might be different from the Alluxio master clock timing, so this check could give us false positives about files that aren't actually changed, or false negatives about files that were loaded to Alluxio and then quickly changed. Even if data is written through Alluxio to the UFS, we may immediately say the data is inconsistent because the UFS clock is ahead of Alluxio's. @calvinjia @maobaolong do you have any ideas on how to address this issue?